packages/js/src/generators/library/library.ts

```diff
@@ -619,5 +627,77 @@ function createProjectTsConfigJson(tree: Tree, options: NormalizedSchema) {
   );
 }
 
+function determineDependencies(
+  options: LibraryGeneratorSchema
+): Record<string, string> {
+  switch (options.bundler) {
+    case 'tsc':
+      // importHelpers is true by default, so need to add tslib as a dependency.
+      return {
+        tslib: tsLibVersion,
+      };
+    case 'swc':
+      // externalHelpers is true  by default, so need to add swc helpers as a dependency.
+      return {
+        '@swc/helpers': swcHelpersVersion,
+      };
+    default: {
+      // In other cases (vite, rollup, esbuild), helpers are bundled so no need to add them as a dependency.
+      return {};
+    }
+  }
+}
+
+type EntryField = string | { [key: string]: EntryField };
+
+function determineEntryFields(
```

Comment:
We'll also include `exports` field and other things in a follow-up PR. We just need to figure out how we want to handle that first.